api/src/test/java/org/openmrs/module/pacsintegration/incoming/ORU_R01ComponentTest.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+
+package org.openmrs.module.pacsintegration.incoming;
+
+import ca.uhn.hl7v2.model.Message;
+import ca.uhn.hl7v2.parser.PipeParser;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.openmrs.Encounter;
+import org.openmrs.Patient;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.emrapi.EmrApiProperties;
+import org.openmrs.module.pacsintegration.api.PacsIntegrationService;
+import org.openmrs.module.radiologyapp.RadiologyProperties;
+import org.openmrs.parameter.EncounterSearchCriteriaBuilder;
+import org.openmrs.test.jupiter.BaseModuleContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.StringContains.containsString;
+
+public class ORU_R01ComponentTest extends BaseModuleContextSensitiveTest {
+
+    protected static final String XML_METADATA_DATASET = "org/openmrs/module/pacsintegration/include/pacsIntegrationTestDataset-metadata.xml";
+    protected static final String XML_MAPPINGS_DATASET = "org/openmrs/module/pacsintegration/include/pacsIntegrationTestDataset-mappings.xml";
+    protected static final String XML_DATASET = "org/openmrs/module/pacsintegration/include/pacsIntegrationTestDataset.xml";
+
+    @Autowired
+    private PatientService patientService;
+
+    @Autowired
+    private EncounterService encounterService;
+
+    @Autowired
+    private EmrApiProperties emrApiProperties;
+
+    @Autowired
+    private RadiologyProperties radiologyProperties;
+
+    @Autowired
+    ORU_R01Handler oruR01Handler;
+
+    @BeforeEach
+    public void setup() throws Exception {
+        executeDataSet(XML_METADATA_DATASET);
+        executeDataSet(XML_MAPPINGS_DATASET);
+        executeDataSet(XML_DATASET);
+        this.getConnection().commit();
+        this.updateSearchIndex();
+        Context.clearSession();
+        Context.getService(PacsIntegrationService.class).initializeHL7Listener();
+        oruR01Handler.setTaskRunner(runnable -> runnable.run());
```

Comment:
By setting the task runner to simply run the method, we are just using the existing context, which works well.